### PR TITLE
Issue 3494: set the deprecrated table option for all Views with grids.

### PR DESCRIPTION
**Pull request by herbdool**
Addresses https://github.com/backdrop/backdrop-issues/issues/3494

### CODE REVIEW

File: core/modules/views/views.install [hunk 1]
```diff
@@ -149,6 +149,33 @@ function views_update_1004() {
...
 
+/**
+ * Set all grid style Views to use the deprecrated table if not set already.
+ */
+function views_update_1005() {
+  $active_storage = config_get_config_storage('active');
+  $names = $active_storage->listAll();
+
+  foreach ($names as $name) {
+    if (substr($name, 0, 11) != 'views.view.') {
```
Comment (klonos) [inline core/modules/views/views.install]: How about using something like:

```php
foreach (config_get_names_with_prefix('views.view.') as $config_name) {
  ... do  ...things
}
```

File: core/modules/views/views.install [hunk 2]
```diff
@@ -149,6 +149,33 @@ function views_update_1004() {
...
+      continue;
+    }
+    $config = config($name);
+    $displays = $config->get('display');
+    foreach ($displays as $display => $display_details) {
+      if (!isset($display_details['display_options']['style_plugin']) || $display_details['display_options']['style_plugin'] != 'grid') {
+        continue;
+      }
+      if (!empty($display_details['display_options']['style_options']['deprecated_table'])) {
```
Comment (klonos) [inline core/modules/views/views.install:165]: ...why not include this in the previous conditional?

File: core/modules/views/views.install [hunk 3]
```diff
@@ -149,6 +149,33 @@ function views_update_1004() {
...
+        continue;
+      }
+      $config->set('display.' . $display . '.display_options.style_options.deprecated_table', '1');
```
Comment (klonos) [inline core/modules/views/views.install:168]: I was always confused as to why we sometimes use `'1'`/`'0'` (which gets saved in the config file as text `"1"`/`"0"`), and other times `1`/`0` (which gets saved in the config file as numeric `1`/`0`): https://github.com/backdrop/backdrop-issues/issues/2425

File: core/modules/views/views.install (hunk 2 above)
Comment (klonos) [reply core/modules/views/views.install:165]: ...apparently after (re)reading http://php.net/manual/en/control-structures.continue.php I don't know how `continue` works, hence the silly questions 😅

File: core/modules/views/views.install (hunk 1 above)
Comment (herbdool) [reply core/modules/views/views.install]: ah yes, that's a better function. Thanks @klonos.

File: core/modules/views/views.install (hunk 3 above)
Comment (herbdool) [reply core/modules/views/views.install:168]: This is common in PHP. Unless the variable is cast to be a specific type, PHP will make a guess.

**Comment by quicksketch** [comment]
Thanks! Merged into 1.x and 1.12.x.

**Pull request merged**